### PR TITLE
Convert even more paths to std::filesystem::path

### PR DESCRIPTION
**Pull request by hexagonrecursion**
* Continuation from #1747

* **Note: this pull request includes** #1778

## Context

In our codebase std::string usually (always?) uses utf-8. Conversions of std::filesystem::path to/from std::string provided by the C++ standard library assume std::string uses "native narrow encoding". This was causing encoding problems on windows.

### CODE REVIEW

**Comment by hexagonrecursion** [comment]
I have eliminated all usage of TempToString() and TempToPath(). I kept the functions themselves in case we want to refactor more paths

File: colobot-base/src/ui/controls/edit.cpp [hunk 1]
```diff
@@ -1617,7 +1616,7 @@ bool CEdit::ReadText(const std::filesystem::path& filename)
             if ( m_bSoluce || !bInSoluce )
             {
                 HyperLink link;
-                link.name = GetNameParam(buffer.data()+i+3, 0);
+                link.name = StrUtils::ToPath(GetNameParam(buffer.data()+i+3, 0));
```
Comment (melex750) [inline colobot-base/src/ui/controls/edit.cpp:1616]: Many SatCom links are broken here most likely due to help files using backslash in file paths.

OS: Xubuntu 24.04.1 LTS

Comment (melex750) [reply colobot-base/src/ui/controls/edit.cpp:1616]: This line from log seems to show backslash as part of the filename:

> [ERROR]: Failed to load text file help/E/cbot\radarall.txt

Comment (hexagonrecursion) [reply colobot-base/src/ui/controls/edit.cpp:1616]: Found it. I removed this line:

https://github.com/colobot/colobot/blob/84ac0d3ba8bc7c590da0b999f132f765f6d56ed1/src/ui/controls/edit.cpp#L819

Will fix soon

**Pull request merged**